### PR TITLE
ARROW-1093: [Python] Run flake8 in Travis CI. Add note about development to README

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

File: ci/travis_script_python.sh [hunk 1]
```diff
@@ -92,7 +92,10 @@ python_version_tests() {
   conda install -y -q nomkl
 
   # Expensive dependencies install from Continuum package repo
-  conda install -y -q pip numpy pandas cython
+  conda install -y -q pip numpy pandas cython flake8
+
+  # Fail fast on style checks
+  flake8 pyarrow
```
Comment (xhochy) [inline ci/travis_script_python.sh]: What about using `pytest-flake8` instead?

Comment (wesm) [inline ci/travis_script_python.sh]: I took a look at this. I figured out how to get flake8 to run on Cython files, and I don't think that can be run within pytest-flake8, so probably simplest to leave this as is (now there's an extra call to check Cython files with some flake8 checks disabled)

File: python/.flake8.cython [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+[flake8]
+filename = *.pyx,*.pxd,*.pxi
+ignore = E211,E901,E225,E226,E227
```
Comment (pcmoritz) [inline python/.flake8.cython]: oops newline

**Comment by pcmoritz** [comment]
Changes look great (I don't like having no space before "\" for line continuation, but it's a very minor detail)

**Comment by wesm** [comment]
I don't have an opinion, so I added spaces before the continuation characters

**Pull request closed without merging**